### PR TITLE
feat(tsSDK): fix constructor registration

### PR DESCRIPTION
**Pull request by TomChv**
Updates entrypoint loading by:
- Loading args type in constructor

Fixes argument loading by:
- Correctly serializing input and parentJSON
- Fixing loadArg to correctly supports list
- Fixing loadArg to load object with ID only
- Refactor load to avoid repetition

Update invoke to:
- Order argument
- Load values, state and result

Update loader to:
- Load arguments order
- Load result
- Adapt to new scanResult data structure

Update scanner to:
- Use a new data structure based on object instead of arrays to increase
performances and maintainability on `ScanResult`.
- Scan non-exposed field
- Add `isExposed` property to `FieldTypeDef`

Update register to:
- Only register exposed field to Dagger API
- Use new data structure scanResult
- Correctly assigning constructor if there's

*Additional notes* 
- Registry is no longer in charge of ordering arguments.
- Update and improve unit tests and integration tests
- Updates entrypoint to no longer loads values but send them 
raw JSON to invoke.

Fixes DEV-3309
Fixes DEV-3314
Fixes DEV-3313

### CODE REVIEW

**Review by helderco — COMMENTED** [review]
Just needs tests.

**Comment by TomChv** [comment]
Requires #6437 to be merged

File: core/integration/module_test.go [hunk 1]
```diff
@@ -3442,6 +3442,57 @@ class Test:
...
+	dir: Directory
+				
+	@field
+	bar: number
+				
+	@field
+	baz: string[]
+				
+	@field
+	neverSetDir?: Directory = undefined
```
Comment (helderco) [inline core/integration/module_test.go:3467]: Just curious, does this work?
```suggestion
	neverSetDir?: Directory
```

**Comment by gerhard** [comment]
Do we want this in today or in the next release @helderco?

I can see that you have left a few comments that have not been addressed yet.

**Comment by helderco** [comment]
> Do we want this in today or in the next release @helderco @TomChv?
> 
> I can see that you have left a few comments that have not been addressed yet.

Main thing is that this depends on another PR that @TomChv as closed to open a new one. If he can get that done in time I'd like to get this in for the release.

File: core/integration/module_test.go [hunk 2]
```diff
@@ -3442,6 +3442,57 @@ class Test:
...
+	}
+				
+	@func
+	gimmeBaz(): string[] {
+		return this.baz;
+	}
+				
+	@func
+	async gimmeDirEnts(): Promise<string[]> {
+		return this.dir.entries();
```
[the remark this answers is not in the document]
Comment (TomChv) [reply core/integration/module_test.go:3505]: It does, `await` is optional on `return`

**Comment by TomChv** [comment]
@helderco I added everything in that PR, feel free to review, I'm waiting for the checks to pass

File: core/integration/module_test.go (hunk 1 above)
Comment (TomChv) [reply core/integration/module_test.go:3479]: It does! I'll update that after your review

File: sdk/typescript/entrypoint/load.ts [hunk 3]
```diff
@@ -79,36 +119,63 @@ export function loadPropertyType(
  * Note: The JSON.parse() is required to remove extra quotes
  */
 export async function loadArg(
-  value: string,
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
```
Comment (helderco) [inline sdk/typescript/entrypoint/load.ts:122]: These line comments are littered everywhere and make the code harder to read. Better to add them to eslint config and only use the inline comments sparsely on an exception to a rule.

File: sdk/typescript/introspector/test/invoke.spec.ts [hunk 4]
```diff
@@ -51,31 +50,59 @@ describe("Invoke typescript function", function () {
...
-      objectName: "Bar",
-      methodName: "exec",
-      state: {},
-      inputs: {
+      parentName: "Bar",
+      fnName: "exec",
+      parentArgs: {},
+      fnArgs: {
         // string[]
         cmd: ["echo", "-n", "hello world"],
```
Comment (helderco) [inline sdk/typescript/introspector/test/invoke.spec.ts:62]: The indentation looks a bit off here. Isn't the linter covering this?

File: sdk/typescript/introspector/test/invoke.spec.ts [hunk 5]
```diff
@@ -51,31 +50,59 @@ describe("Invoke typescript function", function () {
...
+      },
+    }
+
+    // We wrap the execution into a Dagger connection
+    await connection(async () => {
+      const result = await invoke(scanResult, input)
+
+      // We verify the result
+      assert.equal(result, 11)
+    })
```
Comment (helderco) [inline sdk/typescript/introspector/test/invoke.spec.ts:103]: Is the connection really needed? If there's no core types, `.id()` won't be called on any. What's inside invoke that needs it then?

File: sdk/typescript/introspector/test/scan.spec.ts [hunk 6]
```diff
@@ -171,13 +172,13 @@ describe("scan static TypeScript", function () {
 
     const result = scan(files)
     const expected: ScanResult = {
```
Comment (helderco) [inline sdk/typescript/introspector/test/scan.spec.ts:174]: I hope you're not managing this by hand 😱

**Review by helderco — APPROVED** [review]

File: sdk/typescript/introspector/test/invoke.spec.ts (hunk 4 above)
Comment (TomChv) [reply sdk/typescript/introspector/test/invoke.spec.ts:62]: Yes it does :D

File: sdk/typescript/introspector/test/invoke.spec.ts (hunk 5 above)
Comment (TomChv) [reply sdk/typescript/introspector/test/invoke.spec.ts:103]: That's right, but for consistency I preffered to keep the connection

File: sdk/typescript/introspector/test/scan.spec.ts (hunk 6 above)
Comment (TomChv) [reply sdk/typescript/introspector/test/scan.spec.ts:174]: I do :)

File: sdk/typescript/entrypoint/load.ts (hunk 3 above)
Comment (TomChv) [reply sdk/typescript/entrypoint/load.ts:122]: I could disable it for the file! Will think about it in the next PR

**Pull request merged**